tests/recipe_tests/test_HDFReferenceRecipe.py

```diff
@@ -39,15 +39,23 @@ def test_single(netcdf_local_file_pattern_sequential, tmpdir, with_intake):
         )
         m = fs.get_mapper("")
         ds = xr.open_dataset(m, engine="zarr", chunks={}, consolidated=False)
-    assert (ds.foo == expected.foo).all()
+    xr.testing.assert_equal(ds, expected)
```

Comment:
I changed this just to scratch an itch, and interestingly, it turned up a test failure.

```
>       xr.testing.assert_equal(ds, expected)
E       AssertionError: Left and right Dataset objects are not equal
E       Differing dimensions:
E           (time: 10, lat: 18, lon: 36) != (time: 2, lat: 18, lon: 36)
E       Differing coordinates:
E       L * time     (time) datetime64[us] 2010-01-01 2010-01-02 ... 2010-01-10
E       R * time     (time) datetime64[ns] 2010-01-01 2010-01-02
E       Differing data variables:
E       L   foo      (time, lat, lon) float64 0.417 0.7203 0.0001144 ... 0.1179 0.3748
E       R   foo      (time, lat, lon) float64 ...
E       L   bar      (time, lat, lon) float64 9.0 4.0 3.0 2.0 2.0 ... 3.0 3.0 9.0 5.0
E       R   bar      (time, lat, lon) int64 9 4 3 2 2 8 0 0 4 8 ... 1 9 4 5 4 3 1 0 5 3
```

@martindurant - any idea what is going on here?